core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
I know that it had been since forever, but I've never liked the idea that it is single use, imo it's poorly designed and I'd like to change it. The `Interpreter` currently can be re-used, the `EVM` can not. The only reason the `EVM` can not be reused is because of the `abort` field that *may* be set.

Perhaps it is `Context` that should be single use. Argument for this is that you'll never apply a transaction twice and a context is based on a transaction (given it has a field like `Origin`).